### DISCUSSION TITLE
why cant datepicker fields be manually entered?

### DISCUSSION THREAD

**Discussion by MACscr** [filamentphp/filament#363] [Ideas]
Why cant datepicker fields be manually entered? Obviously makes things like birthdates and anything pretty much not within the current year a pain and tedious to set.

**Comment by MACscr** [comment]
No one has any input on this? Im surprised some of these UX issues aren't getting more attention.

**Reply by danharrin** [reply]
If you would like to see changes, please submit a PR or consider funding this work through GitHub Sponsors.

**Filament is free and open source**, it depends on support of the community. Thanks for your interest in it.

**Reply by MACscr** [reply]
My apologies as that wasn't meant as a dig against the project, I am just surprised other people that actually have put Filament into production with any actual end users wouldn't have hit these issues as well that I have brought up recently. And not even for you to solve it, but that it hasn't already been solved by others. Im almost always looking at UX stuff from a client/end user perspective instead of as a developer. So anything that requires thinking/extra clicks from an end users perspective that reduces efficiency of interacting with the data just grinds my gears a bit. Anyway, thanks for your time on this awesome project!

**Reply by blite** [reply]
@MACscr I seriously doubt anyone took it that way.  Its currently almost a whirlwind about how fast changes are happening with this project.  I like everything I see.  

on datepicker I 100% understand what you want.  But I will ask you on datetime fields how would you expect manual entry to work? do you have any good examples?

**Reply by danharrin** [reply]
@blite This is something we've since discussed on Discord, and I believe that @MACscr has yet to find a concrete example of where date pickers with manual entry are used on the web, outside of the native date pickers.

**Reply by blite** [reply]
I've seen quite a few.  I would say this was actually a standard feature a few years ago until people quit including it due to lack of use on mobile.  

Almost always limited to MM/DD/YYYY or YYYY-MM-DD.

This might be the most popular one that supports this https://bootstrap-datepicker.readthedocs.io/en/latest/
https://reactdatepicker.com/ is another that I could easily find. 

This use case really starts to shine on picking date of birth as most of these pickers are extremely slow to type a date of birth. 

For a website that actually supports this I can name one that you use: github milestones, they include a due date field that is typeable, and is actually a great example of how it should work.

**Reply by danharrin** [reply]
I just checked, I'm pretty sure GitHub is using the native HTML date input, which is why it is typeable

**Reply by nettsite** [reply]
Is there a way to use the native HTML date input in a filament form? I feel much like @blite on this, and I need to chat to a client soon whom I know will raise it.

**Reply by nettsite** [reply]
Just found it on a different question:

Alternatively, you can use the native HTML date picker with `TextInput::make()->type('date')`

Also, the year in the Filament date picker popup can be edited (typed), which does help a bit.

**Reply by danharrin** [reply]
In Filament v3, the native date picker is default. You can switch back to the old one with `native(false)`

**Comment by blite** [comment]
I do agree this is needed long term.  Datetime fields also need better entry without 24 hour time.

**Comment by myregistration** [comment]
It would be nice if we could enter input with the non-native DatePicker field

**Comment by FabianoLothor** [comment]
+1

non-native datepicker and timepicker must accept manual input

is the repository opened to receive a PR with that?

**Reply by khalidmaquilang** [reply]
Non-native datepicker(reactive) has an infinite bug when you manually set it up..

> +1
> 
> non-native datepicker and timepicker must accept manual input
> 
> is the repository opened to receive a PR with that?
